### PR TITLE
Fix mismatched Azure event grid identifiable key rule id

### PR DESCRIPTION
**Pull request by yongyan-gh**
# Description

- Update the rule id of `AzureEventGridIdentifiableKey` to `SEC101/199`.
- Update the rule id of `NuGetApiKey` to `SEC101/031`.

### CODE REVIEW

**Comment by yongyan-gh** [comment]
Also the rule id of `NuGetApiKey` is `SEC101/030`, but in the source the id is `SEC101/031`

File: GeneratedRegexPatterns/HighConfidenceSecurityModels.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [
   {
-    "Pattern": "(^|[^abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890+/-_=])(?P<secret>[abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890]{52}JQQJ9(?:9|D|H)[abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890][A-L][abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890]{16}[A-Za-z][abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890]{7}(?:[abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890]{2}==)?)",
+    "Pattern": "(^|[^abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890])(?P<secret>[abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890]{52}JQQJ9(?:9|D|H)[abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890][A-L][abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890]{16}[A-Za-z][abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890]{7}(?:[abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890]{2}==)?)",
```
Comment (nguerrera) [inline GeneratedRegexPatterns/HighConfidenceSecurityModels.json:3]: I'm missing how this relates to the rule id. Is this intentional?

**Review by rwoll — COMMENTED** [review]
Overall looks good but I'm waiting to stamp on resolution of the pending questions about the regex changes as it's not clear how they relate here either.

File: GeneratedRegexPatterns/HighConfidenceSecurityModels.json (hunk 1 above)
Comment (yongyan-gh) [reply GeneratedRegexPatterns/HighConfidenceSecurityModels.json:3]: No, its not related to my changes. These 2 files are autogenerated. 
But the pattern matches the definition in `SEC101_200_CommonAnnotatedSecurityKey`
Looks like out-of-sync change.

Comment (yongyan-gh) [reply GeneratedRegexPatterns/HighConfidenceSecurityModels.json:3]: The latest change of `SEC101_200_CommonAnnotatedSecurityKey` updated the prefix from base64 string to base62, but didn't update the auto-generated file.

image

I think this change is valid.

Comment (nguerrera) [reply GeneratedRegexPatterns/HighConfidenceSecurityModels.json:3]: OK, I'm filing a bug to add PR validation to stop this from getting in like that.

Comment (nguerrera) [reply GeneratedRegexPatterns/HighConfidenceSecurityModels.json:3]: #108

**Review by nguerrera — APPROVED** [review]

**Pull request merged**